Documentation/internal/README.md

```diff
@@ -1,5 +1,160 @@
 # Internal Documentation
 
-TODO(derekparker)
+_This directory holds documentation around the internals of the debugger and how it works._
 
-This directory will hold documentation around the internals of the debugger and how it works.
+This is a general overview of the `delve` internal components.
+
+`delve` consists of the debugger engine which can read and understand Go process memory (located
+in package `proc`), [dwarf](http://dwarfstd.org/doc/dwarf-2.0.0.pdf) 
+[parser](https://golang.org/pkg/debug/dwarf/) and wrapper (go specifics), `terminal` helper,
+`service` set of the RPC servers with user friendly `service/debugger` API.
+
+# `proc` package
+
+The `proc` package is the heart of the `delve` program. It implements low-level debugging interface,
+remote process and memory manipulation, disasm and etc. `proc` is based on the `ptrace(2)` syscall
```

Comment:
`proc` uses ptrace on linux, ptrace + mach specific syscalls on OS X and WaitForDebugEvent+ContinueDebugEvent +SuspendThread+ResumeThread on Windows.